src/operators.jl

```diff
@@ -115,9 +115,19 @@ function union{T<:AbstractGraph}(g::T, h::T)
     return r
 end
 
+isunique(iter::Range) = true
+isunique(iter::Set) = true
+isunique(iter) = (length(unique(iter)) == length(iter))
+
 #@doc "filter g to include only the vertices present in iter which should not have duplicates
 #returns the subgraph of g induced by set(iter) along with the mapping from the old vertex names to the new vertex names" ->
-function inducedsubgraph{T<:AbstractGraph}(g::T, iter)
+function induced_subgraph{T<:AbstractGraph}(g::T, iter)
+    !isunique(iter) && error("Vertices in subgraph list must be unique")
```

Comment:
Why error here instead of using `unique(iter)`? It can't be for performance reasons because you call `unique(iter)` in order to check `isunique`.